feature(classnames-test) added tests for classnames utility function

pravton opened the pull request:
## Description

Added tests cases for classnames utility function which addresses the issue #135 

## Solution

Created a Story with test cases using a Text component and using the classnames directly

## Testing

Checkout this branch
- In one terminal, run cd packages/wethegit-components, followed by yarn start
- In a separate terminal, run yarn test-storybook
- Verify that the tests under the Interactions tab have all passed, respectively
- Additionally, the controls tab can be modified with classnames

marlonmarcello commented:
This is a great start @pravton thank you!

But it's missing quite a few scenarios. Right now you are testing only the "happy path", where the classnames function are getting an array with only strings, which is the "perfect" scenario.

We need to add tests to make sure the classnames function behaves as expected when it also receives things like:
- numbers
- nulls
- undefineds
- objects
- nested arrays
- etc

marlonmarcello commented:
Also, I think you can test the function directly without having to make a component and story for it.

pravton commented:
Thanks @marlonmarcello! I removed the component and updated the test cases. Currently two test cases fail, when pass a number / object, which will get caught with typescript based one the ClassnamesProps but will behave unexpectedly without typescript. What do you think?

marlonmarcello commented on packages/wethegit-components/src/utilities/classnames/classnames.stories.tsx:
```diff
@@ -0,0 +1,43 @@
+    // passing undefined and expecting an empty string
+    expect(classnames(undefined)).toBe("")
+
+    // passing nested array + double nested array and expecting exact result "button"
+    expect(classnames([["button"], [["active"]]])).toBe("button")
+
```
This is interesting, what happens to `active`? We should definitely improve our logic.

pravton replied:
Cool yeah I just added support for the double nested array and now it should return both `"button active"`

marlonmarcello commented:
This is looking really good @pravton !!

> Currently two test cases fail, when passing a number or an object...

I think we need to make sure that the function handle these cases as well @pravton 

Let's improve it!

marlonmarcello commented:
Also, we can't have failing tests otherwise Storybook won't build.

pravton commented:
@marlonmarcello I just improved the logic a bit and now we have passes on all cases. Let me know if anything could be improved.
